### ISSUE TITLE
Script String parameters are not harvested in Java 8

### ISSUE THREAD

**Issue by imagejan** [scijava/scijava-common#175]
I just tried to setup a python script in Fiji using

    # @String name

and discovered that it doesn't work as before: Now an error message is shown:

> 'name' is required but unset.

I tried with any of the "Greeting" templates of the Fiji script editor (Groovy, Javascript, Python, Ruby).
Any idea what has changed?

**Comment by imagejan** [comment]
Sorry, just noticed that this only happens when running Fiji with Java 8 (on Windows).

**Comment by ctrueden** [comment]
Is your Fiji fully up-to-date? I cannot reproduce on OS X.

Does it happen with non-String parameters? I am guessing that Templates > JavaScript > Widgets bombs horribly?

**Comment by imagejan** [comment]
I tried again, now it works with both Java 1.6.0 and Java 1.8.0 under Windows 7.
Sorry for the buzz, I don't know why it happened before.

**Comment by ctrueden** [comment]
@imagejan FWIW, you're not crazy—I have seen that error in the past too, mysteriously. It's been too long now, so I don't remember if I had to make a code change to resolve it, or if it was transient.

Definitely speak up if you see it again!

**Comment by imagejan** [comment]
@ctrueden I was able to reproduce this issue on Mac OSX 10.10.5 after cloning [imagej-ui-swing](https://github.com/imagej/imagej-ui-swing), installing Eclipse (Mars), importing the cloned project, running `ScriptEditorTestDrive.java` from within Eclipse, and opening and running one of the JavaScript templates.
Is it actually supposed to work from within Eclipse? The dependencies look ok to me, but do I have the `SciJavaService`s when running in Eclipse? Pardon my ignorance.

**Comment by ctrueden** [comment]
I tried running `ScriptEditorTestDrive` using the latest master of imagej-ui-swing on my OS X 10.5.5 system, and was unable to reproduce the `'name' is required but unset.` behavior, although it does _not_ work as intended either. In my case, it simply does nothing. No dialog box, no output greeting, but also no error.

It is the end of the day here and I don't have time to debug it now, but that's what would be required to solve it: running in debug mode and digging in to the preprocessors.

What _should_ be happening is that the `SwingInputHarvester` (also of imagej-ui-swing) should be noticing that there is a `String` input parameter, and building a dialog to prompt for it.

> do I have the `SciJavaService`s when running in Eclipse?

In short, yes. But it all depends what is on the classpath at runtime, of course. In the case of imagej-ui-swing, the scijava-ui-swing and scijava-common components are both on the classpath, which contain all the needed bits to make this example work, at least in theory.

**Comment by ctrueden** [comment]
Reopening until we get to the bottom of this...

**Comment by imagejan** [comment]
One possible cause of this might be `ij.plugin.frame.MemoryMonitor` competing with the parameter harvesting, see https://github.com/scijava/scijava-common/issues/241

**Comment by ctrueden** [comment]
I can't reproduce this on macOS nor Windows with an up-to-date Fiji. So I'm going to close this again. See also https://github.com/scijava/scijava-common/issues/241#issuecomment-1091988139. But if it crops again, please feel free to reopen. We might need to debug it directly on the system where it's failing, though.
